### ISSUE TITLE
Document the possibility to force an update of the "alive" state

### ISSUE THREAD

**Issue by AlCalzone** [AlCalzone/node-tradfri-client#107]
See: https://github.com/AlCalzone/node-tradfri-client/issues/106#issuecomment-423179794

**Comment by nykfank** [comment]
1. GET coaps://IP_ADDRESS:5684/15001/DEVICE_ID and extract warmth value from reply
2. PUT warmth - 1 (1 if warmth == 0) to the above URL
3. wait 1 second
4. PUT warmth (original value) to the above URL
5. wait 1 second
6. loop 5 times to point 2

I don't know if there's better method, but this work fine. I used warmth because it doesn't noticeably influence the bulb state if it is powered on. Changing brightness would turn on the bulb, if if is powered on and in OFF state on the gateway. In my tests, the true alive status was determined after 6 seconds. The loop runs to 10 seconds to make sure. I also tried just changing the warmth once, but then it would take full 10 seconds to get the true status.

**Comment by AlCalzone** [comment]
Did you check the gateway updates (observeDevice) for turned-off bulbs? Do those come and include the correct color temperature (the one you just set)? If yes, this could be a way to bail out of the check loop early if the bulb is alive.

**Comment by nykfank** [comment]
Yes, that could be an interesting idea! At what endpoint can I get the gateway updates? 15011/15012?
I would like to use your library, but couldn't get node/npm installed on my SheevaPlug yet.

**Comment by AlCalzone** [comment]
Basically you have set the observe option for the specific endpoint you are interested in, eg devices/deviceID

**Comment by realjax** [comment]
Is it really needed to change a value for a device to receive an update? The warmth, for instance won't work with a non-white spectrum light.  And from what I understood the gateway will also respond with a new alive value after it receives an update for a bulb in which no properties have been changed. 

Am I correct in assuming that if a lightOperation was sent to a device that contains no changes, that this update is not sent?  And if so, maybe implement a force update option?

**Comment by AlCalzone** [comment]
> Is it really needed to change a value for a device to receive an update? The warmth, for instance won't work with a non-white spectrum light.

You can also use the brightness for no-spectrum lights (with the downside that turned-off bulbs may be turned on) or color/hex/sat for RGB lights.
Currently, the way this lib is designed, the bare minimum of data is transmitted. This means that a no-change operation won't send anything. I'll have to add a way to force the command to be sent.

**Comment by realjax** [comment]
> I'll have to add a way to force the command to be sent.

That would be awesome! :)

**Comment by realjax** [comment]
You were right, setting the brightness for no-spectrum lights turns them on, so that is not an option.  I then tried it by changing the transitionTime ( using operateLigh() ) and all though the command (seems) tor work ok, it  does not trigger any callbacks.

So I guess it is safe to say that at the moment for no-spectrum lights there is no way to determine  the "alive" state by forcing an update. ( unless witching the ligh on when it is off is not a problem of course :) )
